Adds Urban Dictionary Plugin

caleywoods opened the pull request:
This adds an Urban Dictionary plugin so users can get definitions from
the Urban Dictionary site via the /ud command to the bot.

caleywoods committed:
Adds Urban Dictionary Plugin

This adds an Urban Dictionary plugin so users can get definitions from
the Urban Dictionary site via the /ud command to the bot.

caleywoods commented:
This pull request is for #126 for Hacktoberfest.

CapacitorSet commented:
The CI failure is ok, I had forgotten to update the test suite and it didn't go into your fork.

Thanks for your contribution! It seems alright, but I'll give it a try this GMT-afternoon and merge it.

CapacitorSet commented on src/plugins/UrbanDictionary.js:
```diff
@@ -0,0 +1,46 @@
+        return {
+            name: "UrbanDictionary",
+            description: "Fetches the Urban Dictionary definition for a supplied word or phrase.",
+            help: ` commands:
```
For consistency, I think it's better if you only include the list of commands, without the "commands:" header.

CapacitorSet commented on src/plugins/UrbanDictionary.js:
```diff
@@ -0,0 +1,46 @@
+                        }
+
+                        this.sendMessage(message.chat.id, msg, opts);
+                    });
```
It lacks a .catch statement: if the request fails for whatever reason, displaying a short error message (possibly with an error on the command line, this.log.error/this.log.warn) is more user-friendly than not showing anything.

caleywoods committed:
Adds Error Checking And Help Response Cleanup

The ajax request to Urban Dictionary now has a catch statement to
display an error to the end user.

"Commands" verbiage was removed from the /help text.

caleywoods commented:
Thanks for "catching" 😄 my missing catch statement. I guess I picked a bad plugin to base my help on in `Quote.js`.

You asked me to use the Wiki and that seemed complete enough for a simple plugin like this. Between the wiki and looking at a few of the existing plugins, as well as the `node-telegram-bot-api` library in `node_modules` to find an example of Promise based requests, was enough to get this working.

caleywoods committed:
Sends Error To Console

This adds error logging out to the bot console.

A bit of text was also re-added to the `/help` command.

caleywoods committed:
JSON.parse Returned Error

JSON parse the returned error so it's readable in the console.

crisbal commented on src/plugins/UrbanDictionary.js:
```diff
@@ -0,0 +1,49 @@
+        return {
+            name: "UrbanDictionary",
+            description: "Fetches the Urban Dictionary definition for a supplied word or phrase.",
+            help: `/ud <query> returns the definition for a given word or phrase`
```
What about using 

```
"`/ud <query>` returns the definition for a given word or phrase"`
```

?

caleywoods committed:
Rework Help Verbiage

Fixes help verbiage a bit.

crisbal commented:
Looks good to me. I just added a minor suggestion on the help message (so the command will be formatted in the help). If you want you can fix it and I (or @CapacitorSet) will merge.

caleywoods commented:
Added your suggestion to the verbiage. Thanks for looking out.
